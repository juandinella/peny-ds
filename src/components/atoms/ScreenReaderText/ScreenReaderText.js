import React from 'react'
import PropTypes from 'prop-types'
import styles from './ScreenReaderText.styles'

/**
 * Hidden text just reachable for screen readers
 */

const ScreenReaderText = ({ id, children }) => {
  return (
    <span css={styles} id={id} data-testid={id}>
      {children}
    </span>
  )
}

ScreenReaderText.displayName = 'ScreenReaderText'

ScreenReaderText.propTypes = {
  /** Component's id */
  id: PropTypes.string,
  /** Component's content */
  children: PropTypes.string.isRequired,
}

ScreenReaderText.defaultProps = {}

export default ScreenReaderText
