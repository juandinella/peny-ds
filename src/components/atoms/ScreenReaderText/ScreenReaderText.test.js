import React from 'react'

import { render, cleanup } from '@testing-library/react'

import ScreenReaderText from './ScreenReaderText'

import initStoryshots from '@storybook/addon-storyshots'

initStoryshots({
  storyKindRegex: /^Atoms\/ScreenReaderText$/,
})

const componentProps = {
  id: 'unique_id',
  children: 'Go!',
}

afterEach(cleanup)

describe('<ScreenReaderText />', () => {
  test('should render component correctly', () => {
    const { queryByTestId } = render(<ScreenReaderText {...componentProps} />)

    const text = queryByTestId(componentProps.id)

    expect(text).toBeTruthy()
  })

  test('should render content correctly', () => {
    const { queryByTestId } = render(<ScreenReaderText {...componentProps} />)

    const text = queryByTestId(componentProps.id)

    expect(text.textContent).toEqual(componentProps.children)
  })
})
