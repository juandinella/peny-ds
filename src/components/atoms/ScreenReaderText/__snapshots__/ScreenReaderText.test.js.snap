// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/ScreenReaderText Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <span
    className="sc-2wwyod-0 jUxFrS"
  >
    Hidden text
  </span>
</div>
`;
